app/assets/javascripts/d3/merit_order_hourly_supply.coffee

```diff
@@ -42,6 +42,14 @@ D3.merit_order_hourly_supply =
 
       Math.abs((d3.max(values) - d3.min(values)) / d3.sum(values))
 
+    getLegendSeries: ->
+      legendSeries = []
+      @series.forEach (serie) =>
+         if d3.max(serie.future_value()) > 0
```

Comment:
This could be...

```js
serie.future_value().find(function(value) { return value > 0 });
```

... which will return upon finding the first positive value (`max` will scan the whole array).